### PR TITLE
feat(core/xref): provide better hints if term is ambiguous

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/w3c/respec/issues/2270
image

### CODE REVIEW

File: src/core/xref.js [hunk 1]
```diff
@@ -391,16 +390,22 @@ function showErrors({ errorsAmbiguous, errorsTermNotFound }) {
...
+    const msg = `The term "**${term}**" is defined in ${specs} in multiple ways, so it's ambiguous.`;
+    let hint = "";
+    if (definedInSpecs.size === 1) {
+      // defined only in one spec but in multiple ways
+      const xrefFor = new Set([].concat(...results.map(entry => entry.for)));
+      const forContext = [...xrefFor].map(s => `**${s}**`).join(", ");
+      hint = `add \`data-xref-for\` attribute with value equal to one of the following: ${forContext}.`;
+    } else {
+      // defined in multiple specs
+      hint = `add ${dataCiteLink} attribute with value equal to one of the following: ${specs}.`;
```
Comment (marcoscaceres) [inline src/core/xref.js]: I wonder if we should mention that they should use the "[= for / term=]" syntax?

Comment (sidvishnoi) [reply src/core/xref.js]: Maybe add wiki link to `data-xref-for` and mention shorthand syntax there?

Comment (marcoscaceres) [reply src/core/xref.js]: Yeah, probably better

Comment (marcoscaceres) [reply src/core/xref.js]: We should move people off using data-*

**Pull request merged**